consumer.go

```diff
@@ -22,6 +22,7 @@ type consumer struct {
 	once        sync.Once
 	messageCh   chan Message
 	quit        chan struct{}
+	apiShutdown chan struct{}
```

Comment:
Do we really need another quit channel? Why don't we call the stop functions of all sub-started processes by listening only to the quit channel?